File: Algorithms/Sum of Square Numbers/sum-of-square-numbers.js
// Source : https://leetcode.com/problems/sum-of-square-numbers/description/
// Author : Han Zichi
// Date   : 2017-07-29

/**
 * @param {number} c
 * @return {boolean}
 */
var judgeSquareSum = function(c) {
  let mid = ~~(Math.sqrt(c))

  for (let i = 0; i <= mid; i++) {
    let rem = c - i * i
    if (rem === Math.pow(~~Math.sqrt(rem), 2))
      return true
  }

  return false
};

File: Algorithms/Sum of Square Numbers/sum-of-square-numbers.py
# Source : https://leetcode.com/problems/sum-of-square-numbers/description/
# Author : Han Zichi
# Date   : 2017-07-29

import math

class Solution:
  def judgeSquareSum(self, c):
    """
    :type c: int
    :rtype: bool
    """
    mid = math.floor(math.sqrt(c))

    for i in range(0, mid + 1):
      rem = c - i * i
      if (rem == math.pow(math.floor(math.sqrt(rem)), 2)):
        return True 

    return False

obj = Solution()
print(obj.judgeSquareSum(5))